### PR TITLE
Order creation: Minor improvements of errors and mandatory information

### LINKED ISSUE #9298: Product creation: Minor improvements of errors and mandatory information
## Description

Note: This issue can be split into several even smaller issues to be solved. Every tiny pull request is welcome!

During testing of #9177 several minor inconsistencies have been found [see comment](https://github.com/openfoodfoundation/openfoodnetwork/pull/9177#issuecomment-1132692337).  Here they are:
- Add "unit value" in the content of the red box at the top of the page if it was left blank during product creation. Currently this is only displayed if a 'unit size' had been selected in the first place.
grafik
- Add red highlighting and little text 'can't be blank' under the fields
  - unit size
  - unit value
  - product category
  grafik
- remove red star on Tax category (default value is preselected)
grafik
- remove red star on Supplier (first supplier is preselected)
grafik
- remove the option to select a blank 'unit size' in the dropdown (in 'product category' dropdown it is not possible to select 'blank')
Peek 2022-06-11 01-46

## Expected Behavior

See above.

## Actual Behaviour

See above.

## Steps to Reproduce

1. Start to create a new product on page /admin/products/new.
2. Leave fields with mandatory information blank (e.g. the product category).
3. Click 'Create'.
4. Observe the error messages and red highlighting.

## Animated Gif/Screenshot

Peek 2022-06-11 01-40

## Severity
bug-s4: it's annoying, but you can use it

## Your Environment

* Version used:
* Browser name and version:
* Operating System and version (desktop or mobile):

### PR DESCRIPTION
**Pull request by uzorjchibuzor**
Closes #9298

* This PR is not ready as I have been unable to remove the blank option from the Unit size dropdown. Angular is not my strong suit.

Progress on the Issue concerned:

-- Add red highlighting and little text 'can't be blank' under the fields
- [ ] unit size
- [x] unit value
- [x] product category
- [x] remove red star on Tax category (default value is preselected)
- [x] remove red star on Supplier (first supplier is preselected)
- [ ] remove the option to select a blank 'unit size' in the dropdown (in 'product category' dropdown it is not possible to select 'blank')

https://user-images.githubusercontent.com/32690770/174094586-b0034b3d-ed2a-4f72-bc6d-ab83b254ca18.mov

### CODE REVIEW

**Comment by drummer83** [comment]
Hey @uzorjchibuzor,
as mentioned in the issue it is absolutely no problem to solve only some of the problems. If you can manage to pass all the tests, this would be a really great contribution already.
After that you or someone else can try to solve the last bit in a separate pull request!

One question: I can see that you added the 'can't be blank' text below the 'Supplier' field as well. How did you manage to have supplier blank? My experience was that the first supplier is always preselected. In that case we wouldn't need the red text below it.

Thanks for your work so far! Looking forward to the final PR!

**Comment by RachL** [comment]
@drummer83 when your account has access to several suppliers, the dropdown is blank by default. Supplier is pre-selected only if your account has access to only one supplier.

**Comment by drummer83** [comment]
@RachL We had this thought already and I understood that the behaviour was changed here: https://github.com/openfoodfoundation/openfoodnetwork/pull/9177#issuecomment-1132725235.
But I can see that the PR had been merged only a few days ago - maybe that's why behaviour in this draft is still the old behaviour. :thinking:

**Comment by uzorjchibuzor** [comment]
I am going to create a few dummy suppliers locally to test out all this
scenarios.

**Comment by uzorjchibuzor** [comment]
I have no idea why these tests are failing, I did not remotely go near any of the instances that I am seeing here. I have checked locally. The tests took more than 30 mins before I stopped it has it was dragging my system but none failed.

File: yarn.lock [hunk 1]
```diff
@@ -2227,23 +2227,6 @@
     ts-dedent "^2.0.0"
     util-deprecate "^1.0.2"
 
-"@storybook/addons@6.5.7":
-  version "6.5.7"
-  resolved "https://registry.yarnpkg.com/@storybook/addons/-/addons-6.5.7.tgz#1c7256f696fb572f7c3c7ccbcb94988372b95cee"
```
Comment (mkllnk) [inline yarn.lock:2232]: These changes don't belong into this pull request. Same with the `dump.rdb` file. Please commit only changes related to the issue you are solving and address only one issue per commit.

File: app/models/spree/product.rb [hunk 2]
```diff
@@ -109,7 +109,7 @@ class Product < ApplicationRecord
                              if: proc { Spree::Config[:products_require_tax_category] }
 
     validates :variant_unit, presence: true
-    validates :unit_value, presence: { if: ->(p) { %w(weight volume).include? p.variant_unit } }
+    validates :unit_value, presence: true
```
Comment (mkllnk) [inline app/models/spree/product.rb:112]: This is not correct. You are changing the validation logic here and this is what is breaking specs. The issue you are working on is only about the feedback the user gets when trying to create a product. It's not about the internal logic.

**Review by mkllnk — COMMENTED** [review]
Hi @uzorjchibuzor, thank you for working on these issues.

I left some comments that may help you solve the problems you were facing. It's a very good idea to run specs locally but as you found out, running all specs takes a long time. Instead you can run only some specs and the command for that is in the spec failure output here in Github Actions. For example:

```
Failed examples:
rspec ./spec/controllers/spree/admin/products_controller_spec.rb:45# Spree::Admin::ProductsController bulk_update when changing a product's variant_unit succeeds
rspec ./spec/controllers/spree/admin/products_controller_spec.rb:58# Spree::Admin::ProductsController bulk_update when changing a product's variant_unit does not redirect to bulk_products
rspec ./spec/controllers/spree/admin/products_controller_spec.rb:99# Spree::Admin::ProductsController bulk_update when passing empty variants_attributes does not fail
```

So you can run this locally and it should be quick:
```
bundle exec rspec ./spec/controllers/spree/admin/products_controller_spec.rb:45
```

Another strategy to find which change is breaking something is to create smaller pull requests. You can create a pull request just with one commit addressing one part of the issue. If the tests pass, we will happily review it. If they fail, you know exactly which code change was wrong.

**Comment by RachL** [comment]
As it's been a long time with no activity here, I'm closing the PR. Anyone is free to reopen if needed!

**Pull request closed without merging**